### PR TITLE
refactor(fixtures): Remove `formats.py`, embed properties as class vars

### PR DESCRIPTION
**Pull request by marioevz**
## 🗒️ Description
~~Minor~~ Major refactor to the `ethereum_test_fixtures` where each subclass of `BaseFixture` now contains all the properties of the format, and therefore `ethereum_test_fixtures/formats.py` is not longer necessary (nor is the Enum it defined).

`FixtureFormats` was widely used across many places in several libraries, hence the high number of files changed, but the changes are relatively minor.

## 🔗 Related Issues
None

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] ~~All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).~~ **skipped**
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened.
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

File: src/pytest_plugins/filler/filler.py [hunk 1]
```diff
@@ -881,9 +877,9 @@ def pytest_generate_tests(metafunc: pytest.Metafunc):
             )
 
 
-def pytest_collection_modifyitems(config, items):
+def pytest_collection_modifyitems(config: pytest.Config, items: List[pytest.Item]):
     """
-    Remove pre-Paris tests parametrized to generate engine type fixtures; these
+    Remove pre-Paris tests parametrized to generate hive type fixtures; these
```
Comment (marioevz) [inline src/pytest_plugins/filler/filler.py:882]:
```suggestion
    Remove pre-Paris tests parametrized to generate engine type fixtures; these
```

**Review by danceratopz — APPROVED** [review]
Nice clean-up! In particular to `fill`'s `pytest_collection_modifyitems()`!

LGTM.

File: src/pytest_plugins/filler/filler.py [hunk 2]
```diff
@@ -892,23 +888,17 @@ def pytest_collection_modifyitems(config, items):
...
-                "state_test"
-            ].name.endswith("ENGINE"):
-                items.remove(item)
-            if ("blockchain_test" in item.callspec.params) and item.callspec.params[
-                "blockchain_test"
-            ].name.endswith("ENGINE"):
-                items.remove(item)
-        if "yul" in item.fixturenames:
+        fork: Fork = params["fork"]
+        if fork.engine_new_payload_version(0, 0) is None:
```
Comment (danceratopz) [inline src/pytest_plugins/filler/filler.py:896]: Just a small helper for the reader.
```suggestion
        # don't generate an "engine" fixture variation pre-merge forks
        if fork.engine_new_payload_version(0, 0) is None:
```

**Review by danceratopz — APPROVED** [review]
Nice clean-up! In particular to fill's `pytest_collection_modifyitems()`! I'd just add a small comment to help the reader.


LGTM.

File: src/pytest_plugins/filler/filler.py (hunk 2 above)
Comment (marioevz) [reply src/pytest_plugins/filler/filler.py:896]: I think we can actually even do something like `fixture_type.supports_fork(fork)` so this is abstracted a bit better.

I'll give it a 5-min try.

Comment (marioevz) [reply src/pytest_plugins/filler/filler.py:896]: > I'll give it a 5-min try.

lmao this prompted a huge refactor.

**Pull request merged**